stix_shifter_modules/nozomi_vantage/stix_transmission/connector.py

```diff
@@ -160,28 +161,24 @@ def handle_metadata_and_page_size(self, offset, length, metadata):
         :return: return_obj, dict
         """
         result_count = 0
-        page_size = 0
         page_number = 1
 
         if metadata:
             if isinstance(metadata, dict) and metadata.get('page_number'):
                 result_count = int(metadata.get('result_count', 0))
                 page_number = int(metadata.get('page_number', 1))
-                page_size = int(metadata.get('page_size', 0))
             else:
                 # raise exception when metadata doesnt contain jwtToken token
                 raise InvalidMetadataException(metadata)
 
         total_records = int(offset) + int(length)
+        page_size = int(length)
 
         if self.api_client.result_limit < total_records:
             total_records = self.api_client.result_limit
 
-        if not metadata:
-            if total_records <= Connector.NOZOMI_MAX_PAGE_SIZE:
-                page_size = total_records
-            else:
-                page_size = Connector.NOZOMI_MAX_PAGE_SIZE
+        if self.api_client.max_page_size < page_size:
```

Comment:
The page size should not change here. There is inherent risk in changing the results by changing the page size. For example, the data on page 100 with page_size = 1 would be very different from a page 100 with page_size = 100. 

I am aware that this works for most situations (having tried it a bit), but this should only be done if there are no other solutions.

Reply:
The length should be constant for all batches.
If the length is less than 1000, page_size will set the lesser value.
If it is greater than 1000, it will be set to 1000.


for ex:
The data source record count is 10000.
First batch
-----------
Input is coming as offset 0, length 500.
Setting page_size to 500, page_number 1
The first iteration fetches 500 records.
so total records return 500.

Second batch
------------
Input is coming as offset 500, length 500.
Setting page_size to 500, page_number 2
The first iteration fetches 500 records.
so total records return 1000.

All the batch lengths are the same. page_size is calculated based on the lenght passed from CP4S. This length is constant batch size hence page_size will not change.